terraform/eks/main.tf

```diff
@@ -109,8 +109,39 @@ locals {
   eks_pod_config = yamldecode(data.template_file.eksconfig.rendered)["sample_app"]
 }
 
+resource "kubernetes_service_account" "aoc-role" {
+  metadata {
+    name = "aoc-role"
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+  }
+
+  automount_service_account_token = true
+}
+
+resource "kubernetes_cluster_role_binding" "aoc-role-binding" {
+  metadata {
+    name = "aoc-role-binding"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "cluster-admin"
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "aoc-role"
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+  }
+}
+
+##########################################
+# Push mode deployments
+##########################################
+
 # deploy aoc and mocked server
 resource "kubernetes_deployment" "aoc_deployment" {
+  count = var.sample_app_mode == "pull" ? 0 : 1
```

Comment:
can we try to run one testcase which is using push mode in case this change break it?

Reply:
I tried running the otlp_mock test case, and it looks like it's passing: 

```
module.validator.null_resource.validator (local-exec): validator_1  | 20:29:56.842 [main] INFO  com.amazon.aoc.validators.MockedServerValidator - mocked server validation passed
module.validator.null_resource.validator (local-exec): eks_validator_1 exited with code 0
module.validator.null_resource.validator (local-exec): Aborting on container exit...
module.validator.null_resource.validator: Creation complete after 1m37s [id=7503739792628168714]
```